### ISSUE TITLE
Collect configuration section in task manifest does not support underscores in metric names

### ISSUE THREAD

**Issue by skonefal** [intelsdi-x/snap#1133]
**OS Version:** Ubuntu 16.04 ```Linux ubuntus 4.4.0-28-generic #47-Ubuntu SMP Fri Jun 24 10:09:13 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux```
**Snap Version:** v0.15.0-beta  `c78752606ed2d27df26ed767a1d6f0b55be0359e`
**Enviroment details:** bare metal

**Steps to reproduce:**
1: Prepare task manifest with metric configuration using underscores:
```
---
  version: 1
  schedule:
    type: "simple"
    interval: "1s"
  max-failures: 10
  workflow:
    collect:
      metrics:
        /intel/metric/under_score: {}
      config:
        /intel/metric/under_score:
          under_score: 750
      publish:
        -
          plugin_name: "file"
          config:
            file: "/tmp/underscore.log"
            debug: true
```
2: Feed it to snapd:
```
snapctl task create -t underscore.yml
```
**Actual Results:**
```
INFO[2016-08-10T11:38:00+02:00] API request                                   _module=_mgmt-rest index=5 method=POST url=/v1/tasks
ERRO[2016-08-10T11:38:00+02:00] Scheduler is not started.                     _block=create-task _module=scheduler error[0]=Invalid namespace: /intel/serenity2/lc/slo/peak_load source=user
INFO[2016-08-10T11:38:00+02:00] API response                                  _module=_mgmt-rest index=5 method=POST status=Internal Server Error status-code=500 url=/v1/tasks
```
**Expected Results:**
Expected workflow to run and configuration passed. Metric with underscores works.

**Comment by IRCody** [comment]
Thank you @skonefal for reporting this.

I was able to successfully reproduce this on the v0.15.0-beta hash provided above (`c78752606ed2d27df26ed767a1d6f0b55be0359e`) and master. The problem is occurring with namespaces in the config section with an under score (which took me a second to notice at first since an underscore in the namespace with no config works fine). The validation on the namespaces provided in the config appears to be overly restrictive/incorrect.

The code for the validation is [here](https://github.com/intelsdi-x/snap/blob/master/scheduler/wmap/wmap.go#L323-L330) and it is called from [here](https://github.com/intelsdi-x/snap/blob/master/scheduler/wmap/wmap.go#L187-L204). 

Part of the work for fixing this would, imo include making the validation of namespaces from metrics and config be identical. 

While looking into your error I also noticed that it was being surfaces under the error "scheduler is not started" which is wrong. This error message is incorrect. I opened issue #1136 to track that.
